src/Compilers/CSharp/Portable/FlowAnalysis/UnassignedFieldsWalker.cs

```diff
@@ -123,6 +123,95 @@ private Symbol GetSymbolForLocation(Symbol symbol)
                 : topLevelMethod;
         }
 
+        public override BoundNode VisitCall(BoundCall node)
+        {
+            var result = base.VisitCall(node);
+            var method = node.Method;
+            if (method.IsStatic || node.ReceiverOpt is BoundThisReference)
+            {
+                ApplyMemberPostConditions(method.ContainingType,
+                    method.NotNullMembers, method.NotNullWhenTrueMembers, method.NotNullWhenFalseMembers);
+            }
+            return result;
+        }
+
+        public override BoundNode VisitPropertyAccess(BoundPropertyAccess node)
+        {
+            var result = base.VisitPropertyAccess(node);
+            var property = node.PropertySymbol;
+            if (property.IsStatic || node.ReceiverOpt is BoundThisReference)
+            {
+                ApplyMemberPostConditions(property.ContainingType,
+                    property.NotNullMembers, property.NotNullWhenTrueMembers, property.NotNullWhenFalseMembers);
+            }
+            return result;
+        }
+
+        private void ApplyMemberPostConditions(
+            TypeSymbol containingType,
+            ImmutableArray<string> notNullMembers,
+            ImmutableArray<string> notNullWhenTrueMembers,
+            ImmutableArray<string> notNullWhenFalseMembers)
+        {
+            if (notNullWhenTrueMembers.IsEmpty && notNullWhenFalseMembers.IsEmpty)
+            {
+                applyMemberPostConditions(notNullMembers, ref State);
+            }
+            else
+            {
+                Split();
+                applyMemberPostConditions(notNullWhenTrueMembers, ref StateWhenTrue);
+                applyMemberPostConditions(notNullWhenFalseMembers, ref StateWhenFalse);
+            }
+
+            void applyMemberPostConditions(ImmutableArray<string> notNullMembers, ref LocalState state)
+            {
+                if (notNullMembers.IsEmpty)
+                {
+                    return;
+                }
+
+                foreach (var notNullMember in notNullMembers)
+                {
+                    markMemberAsAssigned(notNullMember, ref state);
+                }
+            }
+
+            void markMemberAsAssigned(string notNullMember, ref LocalState state)
+            {
+                foreach (Symbol member in containingType.GetMembers(notNullMember))
```

Comment:
>GetMembers(notNullMember [](start = 57, length = 24)

This does not properly handle members declared in the base type and inherited into this type.

Reply:
Stopped reviewing here (Iteration 2).  I will resume review after you have responded to the comments so far.

---
In reply to: [380879807](https://github.com/dotnet/roslyn/pull/41675#discussion_r380879807) [](ancestors = 380879807)

Reply:
@jcouv says "The MemberNotNull attribute can only reference your own/direct fields, but not fields declared in the base. This is consistent with LDM input (ie. use same lookup method that we use for DefaultMember attribute) and I spelled it out more in thread with LDM (and didn't receive any pushback on that particular point)."

---
In reply to: [380881205](https://github.com/dotnet/roslyn/pull/41675#discussion_r380881205) [](ancestors = 380881205,380879807)